diff --git a/src/render/rt64_shader_library.cpp b/src/render/rt64_shader_library.cpp
index d1c5273..c78d87a 100644
--- a/src/render/rt64_shader_library.cpp
+++ b/src/render/rt64_shader_library.cpp
@@ -853,7 +853,7 @@ namespace RT64 {
         }
         
         // Raytracing debug.
-        {
+        if (deviceCapabilities.raytracing) {
             FramebufferRendererDescriptorCommonSet descriptorCommonSet(samplerLibrary, deviceCapabilities.raytracing);
             FramebufferRendererDescriptorTextureSet descriptorTextureSet;
             FramebufferRendererDescriptorFramebufferSet descriptorFramebufferSet;
@@ -866,6 +866,7 @@ namespace RT64 {
             debug.pipelineLayout = layoutBuilder.create(device);
 
             std::unique_ptr<RenderShader> pixelShader = device->createShader(CREATE_SHADER_INPUTS(DebugPSBlobDXIL, DebugPSBlobSPIRV, DebugPSBlobMSL, "PSMain", shaderFormat));
+            pixelShader->setName("RT64 DebugPS raytracing visualization");
             RenderGraphicsPipelineDesc pipelineDesc;
             pipelineDesc.pipelineLayout = debug.pipelineLayout.get();
             pipelineDesc.renderTargetBlend[0] = RenderBlendDesc::AlphaBlend();
